setup.py

```diff
@@ -77,17 +77,19 @@ def get_extras_require() -> Dict[str, List[str]]:
             "mpi4py",
             "mxnet",
             "nbval",
-            "pytorch-ignite",
             "scikit-image",
             "scikit-learn>=0.19.0,<0.23.0",  # optuna/visualization/param_importances.py.
-            "thop",
-            "torch==1.5.1" if sys.platform == "darwin" else "torch==1.5.1+cpu",
-            "torchvision==0.6.1" if sys.platform == "darwin" else "torchvision==0.6.1+cpu",
             "xgboost",
             "keras",
             "tensorflow>=2.0.0",
             "tensorflow-datasets",
         ]
+        + (
+            ["torch==1.6.0" if sys.platform == "darwin" else "torch==1.6.0+cpu"]
+            + ["pytorch-ignite", "thop"]
```

Comment:
[just a question]
Why are you adding `thop` only when the version of Python is greater than 3.5?
[`pytorch_simple.py`](https://github.com/optuna/optuna/blob/4a4d37a6645f841662e5fe57cafef03269b03a0c/examples/multi_objective/pytorch_simple.py#L13) doesn't work on Python3.5?

Reply:
Also curious about this change. Do you have any comments on it @crcrpar?

Reply:
Because `thop` obviously depends on PyTorch https://github.com/Lyken17/pytorch-OpCounter/blob/master/setup.py#L14.

Reply:
Thank you for the explanation.

1. `thop` depends on `PyTorch` explicitly
2. `PyTorch` dropped the support of Python 3.5
3. If `thop` appears at the top of `setup.py`, CIs using Python 3.5 will fail (so we have to conditionally add into dependencies)

It makes sense for me. 👍